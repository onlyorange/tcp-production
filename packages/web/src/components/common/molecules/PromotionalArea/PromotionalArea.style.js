import { css } from 'styled-components';

export default css`
  font-size: ${props => props.theme.fonts.fontSize.body.bodytext.copy1}px;

  .header-topnav__promo-area--mobile {
    background-color: ${props => props.theme.colors.PRIMARY.NAVY};
    color: ${props => props.theme.colors.WHITE};
    font-family: ${props => props.theme.fonts.secondaryFontFamily};
    padding: 8px 0;

    @media ${props => props.theme.mediaQuery.medium} {
      display: none;
    }
  }

  .header-topnav__promo-area--tablet {
    font-family: ${props => props.theme.fonts.secondaryFontFamily};
    margin-top: 16px;

    @media ${props => props.theme.mediaQuery.smallMax} {
      display: none;
    }

    @media ${props => props.theme.mediaQuery.large} {
      font-family: ${props => props.theme.fonts.secondaryFontSemilBoldFamily};
      font-size: ${props => props.theme.fonts.fontSize.body.bodytext.copy4}px;
    }
  }
`;
