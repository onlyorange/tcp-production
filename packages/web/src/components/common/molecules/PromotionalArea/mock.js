export default {
  promo_message_wrapper: [
    {
      id: '1',
      url: '/static/promo',
      text: '<b>FREE SHIPPING EVERY DAY!</b> No minimum purchase required 1.',
      title: 'FREE SHIPPING EVERY DAY! No minimum purchase required 1.',
      target: '_blank',
      class: 'promo_message',
    },
    {
      id: '2',
      url: '/static/promo2',
      text: '<b>FREE SHIPPING EVERY DAY!</b> No minimum purchase required 2.',
      title: 'FREE SHIPPING EVERY DAY! No minimum purchase required 2.',
      target: '_blank',
      class: 'promo_message',
    },
    {
      id: '3',
      url: '/static/promo3',
      text: '<b>FREE SHIPPING EVERY DAY!</b> No minimum purchase required 3.',
      title: 'FREE SHIPPING EVERY DAY! No minimum purchase required 3.',
      target: '_blank',
      class: 'promo_message',
    },
  ],
};
