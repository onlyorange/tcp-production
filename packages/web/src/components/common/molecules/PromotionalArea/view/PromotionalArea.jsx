import React from 'react';
import PropTypes from 'prop-types';
import withStyles from '@tcp/core/src/components/common/hoc/withStyles';
import Carousel from '@tcp/core/src/components/common/molecules/Carousel';
import RichText from '@tcp/core/src/components/common/atoms/RichText';
import CarouselConfig from '@tcp/web/src/config/carousel';
import style from '../PromotionalArea.style';

const PromotionalArea = ({ className, data, mobile }) => {
  const carouselConfig = mobile
    ? CarouselConfig.CAROUSEL_OPTIONS
    : CarouselConfig.CAROUSEL_FADE_OPTIONS;
  const wrapperClass = mobile
    ? 'header-topnav__promo-area--mobile'
    : 'header-topnav__promo-area--tablet';

  return (
    <div className={className}>
      <div className={wrapperClass}>
        <Carousel options={carouselConfig} carouselTheme="dark">
          {data.map(promotion => {
            return <RichText richTextHtml={promotion.text} dataLocator="global_promoareaimg" />;
          })}
        </Carousel>
      </div>
    </div>
  );
};

PromotionalArea.propTypes = {
  className: PropTypes.string.isRequired,
  data: PropTypes.objectOf(PropTypes.oneOfType([PropTypes.number, PropTypes.string])).isRequired,
  mobile: PropTypes.bool.isRequired,
};

export default withStyles(PromotionalArea, style);
export { PromotionalArea as PromotionalAreaVanilla };
