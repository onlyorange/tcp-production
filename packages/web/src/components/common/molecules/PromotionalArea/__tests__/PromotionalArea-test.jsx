import React from 'react';
import { shallow } from 'enzyme';
import PromotionalArea from '../view';
import mockData from '../mock';

describe('PromotionalArea component', () => {
  it('renders correctly', () => {
    const props = {
      data: mockData.promo_message_wrapper,
    };
    const component = shallow(<PromotionalArea {...props} />);
    expect(component).toMatchSnapshot();
  });
});
