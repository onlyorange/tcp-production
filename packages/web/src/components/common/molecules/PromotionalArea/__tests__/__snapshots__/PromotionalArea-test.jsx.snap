// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PromotionalArea component renders correctly 1`] = `ShallowWrapper {}`;
